api/src/adler32.rs

```diff
@@ -0,0 +1,23 @@
+//! # Adler 32
+//!
+//! Adler 32 is checksum algorithm
+//!
+//! [More about Adler-32](https://en.wikipedia.org/wiki/Adler-32)
+
+const ADLER32_MOD: u32 = 0xFFF1;
+
+/// Checksum hash with [Adler-32](https://en.wikipedia.org/wiki/Adler-32)
+#[inline(always)]
+pub fn adler32_checksum(hash: Vec<u8>, init_left: u32, init_right: u32) -> u32 {
```

Comment:
Why not to try somehting more idiomatic:
```rust
pub fn adler32_checksum(hash: Vec<u8>, left: u32, right: u32) -> u32 {
    let (lo, hi) = hash.iter().fold((left, right), |&byte| {
        ((left + byte as u32) % ADLER32_MODE, ( 2 * right + byte as u32) % ADLER32_MOD)
    })
    (hi << 16) | lo
}
```

Also isn't there a mystyped `+=` sign. Looking at the[example implementation](https://en.wikipedia.org/wiki/Adler-32#Example_implementation) it should be something like this:

```rust
pub fn adler32_checksum(hash: Vec<u8>, left: u32, right: u32) -> u32 {
    let (lo, hi) = hash.iter().fold((left, right), |&byte| {
        ((left + byte as u32) % ADLER32_MODE, ( right + left + byte as u32) % ADLER32_MOD)
    })
    (hi << 16) | lo
}
```

Or in more readable version:
```rust
pub fn adler32_checksum(hash: Vec<u8>, left: u32, right: u32) -> u32 {
    let (lsb, msb) = hash.iter().fold((left, right), |&byte| {
        let next_left = mod_sum(left, byte);
        (next_left, mod_sum(right, next_left))
    })
    (msb << 16) | lsb
}

/// Sum two elements in modular arithmetics with module ADLER32_MODE
fn mod_sum(a: u32, b: u32) -> u32 {  // Can be generified
    (a + b) % ADLER32_MODE
}
```

Reply:
> Also isn't there a mystyped += sign. Looking at the[example implementation](https://en.wikipedia.org/wiki/Adler-32#Example_implementation) it should be something like this:

Yeah, I mistyped symbol `+`, Thanks for report. Also I implement test case for adler32 from wiki, just in case if something similar happens again.

Definitely more idiomatic way is much better and shorter. But I have to change some things for my newer user checksum use case, so check it up (づ｡◕‿‿◕｡)づ